python/cudf/cudf/_lib/strings/combine.pyx

```diff
@@ -135,28 +81,12 @@ def join_lists_with_column(
     `<NA>`/`None` values in `separator_strings` are replaced
     by `py_separator_narep`
     """
-
-    cdef DeviceScalar source_narep = py_source_narep.device_value
-    cdef DeviceScalar separator_narep = py_separator_narep.device_value
-
-    cdef unique_ptr[column] c_result
-    cdef column_view source_view = source_strings.view()
-    cdef column_view separator_view = separator_strings.view()
-
-    cdef const string_scalar* scalar_source_narep = \
-        <const string_scalar*>(source_narep.get_raw_ptr())
-    cdef const string_scalar* scalar_separator_narep = <const string_scalar*>(
-        separator_narep.get_raw_ptr()
+    plc_column = plc.strings.combine.join_list_elements(
+        source_strings.to_pylibcudf(mode="read"),
+        separator_strings.to_pylibcudf(mode="read"),
+        py_separator_narep.device_value.c_value,
+        py_source_narep.device_value.c_value,
```

Comment:
*question*: does this work? It seems like one needs to produce a pylibcudf `Scalar`, but this is not that?

Reply:
Ah, ok, I missed that.